src/util/include/util/DispatchPool.h

```diff
@@ -0,0 +1,80 @@
+/*
+ * Xournal++
+ *
+ * Template class for a dispatch pool
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
+ *
+ * @license GNU GPLv2 or later
+ */
+
+#pragma once
+
+#include <algorithm>
+#include <cassert>
+#include <memory>
+#include <vector>
+
+namespace xoj::util {
+template <class T>
+class DispatchPool final {
```

Comment:
Needs a doc comment.

I initially found the type parameter name `T` to be confusing, until I saw the `using listener_type = T` line below. I would suggest renaming `T` to something more understandable like `ListenerT`.